src/esp32/esp32_wifi.c

```diff
@@ -627,11 +630,24 @@ bool mgos_wifi_dev_start_scan(void) {
 
 esp_err_t esp32_wifi_protocol_setup(wifi_interface_t ifx, const char *prot) {
   uint8_t protocol = 0;
-  if (strchr(prot, 'B') != NULL) protocol |= WIFI_PROTOCOL_11B;
-  if (strchr(prot, 'G') != NULL) protocol |= WIFI_PROTOCOL_11G;
-  if (strchr(prot, 'N') != NULL) protocol |= WIFI_PROTOCOL_11N;
-  if (strstr(prot, "LR") != NULL) protocol |= WIFI_PROTOCOL_LR;
-  LOG(LL_INFO, ("WiFi %s: protocol %s (%#x)",
-                (ifx == WIFI_IF_STA ? "STA" : "AP"), prot, protocol));
+  if (strcmp(prot, "B") == 0) {
+    protocol = WIFI_PROTOCOL_11B;
+  } else if (strcmp(prot, "BG") == 0) {
+    protocol = (WIFI_PROTOCOL_11B | WIFI_PROTOCOL_11G);
+  } else if (strcmp(prot, "BGN") == 0) {
+    protocol = (WIFI_PROTOCOL_11B | WIFI_PROTOCOL_11G | WIFI_PROTOCOL_11N);
+  } else if (strcmp(prot, "LR") == 0) {
+    protocol = WIFI_PROTOCOL_LR;
+  } else if (strcmp(prot, "BLR") == 0) {
+    protocol = (WIFI_PROTOCOL_11B | WIFI_PROTOCOL_LR);
+  } else if (strcmp(prot, "BGLR") == 0) {
+    protocol = (WIFI_PROTOCOL_11B | WIFI_PROTOCOL_11G | WIFI_PROTOCOL_LR);
+  } else if (strcmp(prot, "BGNLR") == 0) {
+    protocol = (WIFI_PROTOCOL_11B | WIFI_PROTOCOL_11G | WIFI_PROTOCOL_11N | WIFI_PROTOCOL_LR);
+  } else {
+    //Defaults to BGN
```

Comment:
error out if the value is specified and invalid

Reply:
Done, however this will return `Failed to set STA protocol: ESP_ERR_NOT_SUPPORTED` which then causes a reboot / OTA fail. Not sure this is better than defaulting back to BGN and therefore loading OK?